packages/cli/src/databases/repositories/authUser.repository.ts

```diff
@@ -1,10 +1,68 @@
 import { Service } from 'typedi';
-import { DataSource, Repository } from '@n8n/typeorm';
+import { DataSource, Repository, In, IsNull, Not } from '@n8n/typeorm';
+import type { DeepPartial, EntityManager, FindOptionsWhere } from '@n8n/typeorm';
+
 import { AuthUser } from '../entities/AuthUser';
+import { Project } from '../entities/Project';
+import { ProjectRelation } from '../entities/ProjectRelation';
 
 @Service()
 export class AuthUserRepository extends Repository<AuthUser> {
 	constructor(dataSource: DataSource) {
 		super(AuthUser, dataSource.manager);
 	}
+
+	/** This returns a user object with all columns, including the ones marked as `select: false` */
+	async findForAuth(where: FindOptionsWhere<AuthUser>, includeRelations = false) {
+		return await this.findOne({
+			where,
+			relations: includeRelations ? ['authIdentities'] : undefined,
+		});
+	}
+
+	async findManyByIds(userIds: string[]) {
+		return await this.find({
+			where: { id: In(userIds) },
+		});
+	}
+
+	async createUserWithProject(
+		user: DeepPartial<AuthUser>,
+		transactionManager?: EntityManager,
+	): Promise<{ user: AuthUser; project: Project }> {
```

Comment:
Should this return regular `User`s? Or should we have separate methods for creating member users and creating auth users. I see a risk that these created users are passed on as is and might expose the auth props as well